### PR TITLE
Replace distutils.sysconfig.get_python_lib with sysconfig.get_path

### PR DESCRIPTION
**Pull request by chigozienri**
Fixes https://github.com/enthought/traits/issues/1449

Removes the dependence on distutils, by replacing `distutils.sysconfig.get_python_lib` with `sysconfig.get_path`

`distutils.sysconfig.get_python_lib` with no flags defaults to `plat_specific=0, standard_lib=0`
https://github.com/pypa/distutils/blob/main/distutils/sysconfig.py

This is equivalent to `sysconfig.get_path('purelib')`
https://docs.python.org/3/library/sysconfig.html?highlight=purelib

**Checklist**
- [ n/a ] Tests
- [ n/a ] Update API reference (`docs/source/traits_api_reference`)
- [ n/a ] Update User manual (`docs/source/traits_user_manual`)
- [ n/a ] Update type annotation hints in `traits-stubs`

### CODE REVIEW

File: traits/util/resource.py [hunk 1]
```diff
@@ -30,8 +30,7 @@
 import os
 import sys
 
-from distutils.sysconfig import get_python_lib
-
+from sysconfig import get_path
```
Comment (rahulporuri) [inline traits/util/resource.py:34]:
```suggestion


```

**Review by rahulporuri — APPROVED** [review]
LGTM. I tried this out locally and it looks correct. I'm not sure this code path is tested by the docs though.

File: traits/util/resource.py [hunk 2]
```diff
@@ -30,7 +30,7 @@
 import os
 import sys
 
-from distutils.sysconfig import get_python_lib
+from sysconfig import get_path
```
Comment (rahulporuri) [inline traits/util/resource.py:33]:
```suggestion
from sysconfig import get_path as sysconfig_get_path
```

File: traits/util/resource.py [hunk 3]
```diff
@@ -144,8 +144,8 @@ def find_resource(project, resource_path, alt_path=None, return_path=False):
     except:
         # Setuptools was either not installed, or it failed to find the file.
         # First check to see if the package was installed using egginst by
-        # looking for the file at: site-packages\\resouce_path
-        full_path = os.path.join(get_python_lib(), resource_path)
+        # looking for the file at: site-packages\\resource_path
+        full_path = os.path.join(get_path('purelib'), resource_path)
```
Comment (rahulporuri) [inline traits/util/resource.py:148]:
```suggestion
        full_path = os.path.join(sysconfig_get_path('purelib'), resource_path)
```

**Comment by mdickinson** [comment]
LGTM as a quick fix, though I think we need to revisit `find_resource` and its uses.

I took the liberty of making a drive-by style fix, changing the `from sysconfig import` to a plain `import sysconfig`. (Very much a personal thing, but my preference is to use plain `import` for std. lib. modules where there's no reason to do otherwise.)

**Pull request merged**